scalding-macros/src/test/scala/com/twitter/scalding/macros/MacroOrderingProperties.scala

```diff
@@ -0,0 +1,540 @@
+
+  def gen[T: Arbitrary]: Gen[T] = implicitly[Arbitrary[T]].arbitrary
+
+  def arbMap[T: Arbitrary, U](fn: T => U): Arbitrary[U] = Arbitrary(gen[T].map(fn))
+
+  def collectionArb[C[_], T: Arbitrary](implicit cbf: collection.generic.CanBuildFrom[Nothing, T, C[T]]): Arbitrary[C[T]] = Arbitrary {
+    gen[List[T]].map { l =>
+      val builder = cbf()
+      l.foreach { builder += _ }
+      builder.result
+    }
+  }
+
+  def serialize[T](t: T)(implicit orderedBuffer: OrderedSerialization[T]): InputStream =
+    serializeSeq(List(t))
+
+  def serializeSeq[T](t: Seq[T])(implicit orderedBuffer: OrderedSerialization[T]): InputStream = {
+    import JavaStreamEnrichments._
+
+    val baos = new ByteArrayOutputStream
+    t.foreach({ e =>
+      orderedBuffer.write(baos, e)
+    })
+    baos.toInputStream
+  }
+
+  def rt[T](t: T)(implicit orderedBuffer: OrderedSerialization[T]) = {
+    val buf = serialize[T](t)
+    orderedBuffer.read(buf).get
+  }
+
+  def rawCompare[T](a: T, b: T)(implicit obuf: OrderedSerialization[T]): Int =
+    obuf.compareBinary(serialize(a), serialize(b)).unsafeToInt
+
+  def checkManyExplicit[T](i: List[(T, T)])(implicit obuf: OrderedSerialization[T]) = {
+    val serializedA = serializeSeq(i.map(_._1))
+    val serializedB = serializeSeq(i.map(_._2))
+    i.foreach {
+      case (a, b) =>
+        val compareBinary = obuf.compareBinary(serializedA, serializedB).unsafeToInt
+        val compareMem = obuf.compare(a, b)
+        if (compareBinary < 0) {
+          assert(compareMem < 0, s"Compare binary: $compareBinary, and compareMem : $compareMem must have the same sign")
+        } else if (compareBinary > 0) {
+          assert(compareMem > 0, s"Compare binary: $compareBinary, and compareMem : $compareMem must have the same sign")
+        }
+    }
+  }
+
+  def checkMany[T: Arbitrary](implicit obuf: OrderedSerialization[T]) = forAll { i: List[(T, T)] =>
+    checkManyExplicit(i)
+  }
+
+  def clamp(i: Int): Int =
+    i match {
+      case x if x < 0 => -1
+      case x if x > 0 => 1
+      case x => 0
+    }
+
+  def checkWithInputs[T](a: T, b: T)(implicit obuf: OrderedSerialization[T]) {
+    val rta = rt(a) // before we do anything ensure these don't throw
+    val rtb = rt(b) // before we do anything ensure these don't throw
+    val asize = Serialization.toBytes(a).length
+    if (obuf.dynamicSize(a).isDefined) {
+      assert(obuf.dynamicSize(a).get == asize, "dynamic size matches the correct value")
+    }
+    if (obuf.staticSize.isDefined) {
+      assert(obuf.dynamicSize(a).get == asize, "dynamic size matches the correct value")
+      assert(obuf.staticSize.get == asize, "dynamic size matches the correct value")
+    }
+    assert(oBufCompare(rta, a) === 0, s"A should be equal to itself after an RT -- ${rt(a)}")
+    assert(oBufCompare(rtb, b) === 0, s"B should be equal to itself after an RT-- ${rt(b)}")
+    assert(oBufCompare(a, b) + oBufCompare(b, a) === 0, "In memory comparasons make sense")
```

Comment:
Yeah, so that was my point:
`assert(oBufCompare(a, b) + oBufCompare(b, a) === 0`
is too strict right?
Should be:
`assert(oBufCompare(a, b).signum + oBufCompare(b, a).signum === 0`

unless these compare methods always wind up going through `unsafeToInt` ?